specs/overview.md

```diff
@@ -0,0 +1,158 @@
+
+Both deposit types are represented by a custom EIP-2718 transaction type on L2.
+
+### Block Derivation
+
+#### Overview
+
+The rollup chain can be deterministically derived given an L1 Ethereum chain. The fact that the entire rollup chain can be derived based on L1 blocks is _what makes OE a rollup_. This process can be represented as:
+
+```
+derive_rollup_chain(l1_blockchain) -> rollup_blockchain
+```
+
+Optimism's block derivation function is designed such that it:
+
+- Requires no state other than what is easily accessible using L1 and L2 execution engine APIs.
+- Supports sequencers and sequencer consensus.
+- Is resilient to sequencer censorship.
+
+#### Epochs and the Sequencing Window
+
+The rollup chain is subdivided into epochs. There is a 1:1 correspondence between L1 block numbers and epoch numbers. For L1 block number `n`, there is a corresponding rollup epoch `n` which can only be derived after L1 block number `n` is added to the L1 chain. 
+
+Each epoch contains at least one block. Every block in the epoch contains an L1 info transaction which contains contextual about L1 such as the block hash and timestamp. The first block in the epoch also contains all deposits initiated via the `DepositFeed` contract on L1. All subsequent blocks contain batches of transactions submitted directly to the sequencer.
+
+The sequencer can create blocks for future epochs, and reference them before confirmation on L1. This is what enables _fast transaction confirmations_ on the sequencer. To prevent censorship, however, the sequencer must batch-submit the blocks belonging to each epoch within a fixed number of L1 blocks. This submission window is called the _sequencing window_. The `epoch`s described above correspond to the first block in each sequencing window. Since transaction batches for a given epoch can be submitted anywhere within the sequencing window, verifiers must search all blocks within the window for transaction batches.
+
+The following diagram describes this relationship, and how L2 blocks are derived from L1 blocks (L1 info transactions have been elided):
+
+![Epochs and Sequencing Windows](./assets/sequencer-block-gen.svg)
+
+#### Block Derivation Loop
+
+A sub-component of the rollup node called the _rollup driver_ is actually responsible for performing block derivation. The rollup driver is essentially an infinite loop that runs the block derivation function. For each epoch, the block derivation function performs the following steps:
+
+1. Downloads deposit and transaction batch data for each block in the sequencing window.
+2. Converts the deposit and transaction batch data into payload attributes for the Engine API.
+3. Submits the payload attributes to the Engine API, where they are converted into blocks and added to the canonical chain.
+
+This process is then repeated with incrementing epohcs until the tip of L1 is reached.
+
+### Engine API
```

Comment:
In the overview for infra provider, you call the execution engine "bedrock geth". I kinda like it, do we want to introduce this terminology here / in the specs? (In addition to "execution engine", which we've used a lot).

Reply:
I think we should keep Execution Engine since it's the technically correct term and matches what the rest of the community uses.